pkg/services/mysqldb/bind.go

```diff
@@ -94,3 +178,42 @@ func (s *serviceManager) GetCredentials(
 		Password: bd.Password,
 	}, nil
 }
+
+func (v *vmOnlyManager) GetCredentials(
+	_ service.Instance,
+	_ service.Binding,
+) (service.Credentials, error) {
+	return nil, fmt.Errorf("service not bindable")
+}
+
+func (d *dbOnlyManager) GetCredentials(
+	instance service.Instance,
+	binding service.Binding,
+) (service.Credentials, error) {
+	pdt, ok := instance.Parent.Details.(*vmOnlyMysqlInstanceDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting instance.Parent.Details as *vmOnlyMysqlInstanceDetails",
+		)
+	}
+	dt, ok := instance.Details.(*dbOnlyMysqlInstanceDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting instance.Details as *dbOnlyMysqlInstanceDetails",
+		)
+	}
+	bd, ok := binding.Details.(*mysqlBindingDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting binding.Details as *mysqlBindingDetails",
+		)
+	}
+
+	return &Credentials{
+		Host:     pdt.FullyQualifiedDomainName,
+		Port:     3306,
+		Database: dt.DatabaseName,
+		Username: fmt.Sprintf("%s@%s", bd.LoginName, pdt.ServerName),
+		Password: bd.Password,
+	}, nil
+}
```

Comment:
I know we didn't do this for sqldb, but perhaps should have (and can address in a separate pr), but what would you think about breaking this file (and other similar files) up into three that are each prefixed with the manager name? I feel like the code would be easier to follow (for the purposed of review, for instance) if code for the three managers weren't interspersed in one common set of files.

Reply:
Yeah! I thought we should have done it in sqldb, so this is cool with me!